File: cpp/2352_equal_row_and_column_pairs.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int equalPairs(vector<vector<int>>& grid) {
		map<vector<int>, int> dict;
		for (auto row : grid) {
			dict[row] += 1;
		}

		int ans = 0;
		for (int j = 0; j < grid.size(); j++) {
			vector<int> cur;
			for (int i = 0; i < grid.size(); i++) {
				cur.push_back(grid[i][j]);
			}
			ans += dict[cur];
		}
		return ans;
    }
};

File: python/2352_equal_row_and_column_pairs.py
from typing import List

class Solution:
    def equalPairs(self, grid: List[List[int]]) -> int:

        map = {}
        for row in grid:
            if str(row) not in map:
                map[str(row)] = 0
            map[str(row)] += 1

        ans = 0
        for i in range(len(grid)):
            cur = []
            for j in range(len(grid)):
                cur.append(grid[j][i])
            if str(cur) in map:
                ans += map[str(cur)]

        return ans
